src/test/java/io/airlift/slice/TestSlices.java

```diff
@@ -67,6 +79,58 @@ private static void testWrapping(ByteBuffer buffer)
         }
     }
 
+    @Test
+    public void testWrapAllTypes()
+    {
+        boolean[] booleanArray = {true, false, false, true, false};
+        assertEquals(wrappedBooleanArray(booleanArray).getByte(0) == 1, booleanArray[0]);
+        SliceInput input = wrappedBooleanArray(booleanArray).getInput();
+        input.readBoolean();
+        assertEquals(input.readBoolean(), booleanArray[1]);
```

Comment:
These `SliceInput` tests doesn't seem relevant to the changes your made. In addition, these cases are covered in `AbstractSliceInputTest` already.